magicctapipe/scripts/lst1_magic/lst_m1_m2_cloud_correction.py

```diff
@@ -0,0 +1,417 @@
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def model0(imp, h, zd):
+    """
+    Calculates distance
+
+    Parameters
+    ----------
+    imp : float
+        Impact
+    h : float
+        Height
+    zd : float
+        Zenith distance
+
+    Returns
+    -------
+    float
+        Angular distance in units of degree
+    """
+    d = h / np.cos(zd)
+    return np.arctan((imp / d).to("")).to_value("deg")
+
+
+def model2(imp, h, zd):
+    """
+    Calculates model
+
+    Parameters
+    ----------
+    imp : float
+        Impact
+    h : float
+        Height
+    zd : float
+        Zenith distance
+
+    Returns
+    -------
+    float
+        Bias
+    """
+    H0 = 2.2e3 * u.m
+    bias = 0.877 + 0.015 * ((h + H0) / (7.0e3 * u.m))
+    return bias * model0(imp, h, zd)
+
+
+def trans_height(x, Hc, dHc, trans):
+    """
+    Calculates height of a cloud transmission
+
+    Parameters
+    ----------
+    x : float
+        Position of each layer
+    Hc : float
+        Cloud above the telescope
+    dHc : float
+        Cloud thickness
+    trans : float
+        Transmission of the cloud
+
+    Returns
+    -------
+    float
+        Height of a cloud transmission
+    """
+    t = pow(trans, ((x - Hc) / dHc).to_value(""))
+    t = np.where(x < Hc, 1, t)
+    t = np.where(x > Hc + dHc, trans, t)
+    return t
+
+
+def process_telescope_data(
+    input_file, output_file, telid, focal_eq, focal_eff, camgeom
+):
+    """
+    Corrects LST-1 and MAGIC data affected by a cloud presence
+
+    Parameters
+    ----------
+    input_file : str
+        Path to an input .h5 DL1 file
+    output_file : str
+        Path to a directory where to save an output corrected DL1 data file
+    telid : int
+        LST-1 and MAGIC telescope ids
+    focal_eq : float
+        Equivalent focal length
+    focal_eff : float
+        Effective focal length
+    camgeom : int
+        Camera geometry
+
+    Returns
+    -------
+    pandas.core.frame.DataFrame
+        Data frame of corrected DL1 parameters
+    """
+
+    Hc = 5900 * u.m  # cloud above the telescope
+    dHc = 1320 * u.m
+
+    data = []
+    dl1_params = read_table(input_file, "/events/parameters")
+    dl1_images = read_table(input_file, "/events/dl1/image_" + str(telid))
+
+    inds = np.where(
+        np.logical_and(dl1_params["intensity"] > 0.0, dl1_params["tel_id"] == telid)
+    )[0]
+    for index in inds:
+        event_id_lst = dl1_params["event_id_lst"][index]
+        obs_id_lst = dl1_params["obs_id_lst"][index]
+        event_id = dl1_params["event_id_magic"][index]
+        obs_id = dl1_params["obs_id_magic"][index]
+        if telid == 1:
+            event_id, obs_id = event_id_lst, obs_id_lst
+
+        pointing_az = dl1_params["pointing_az"][index]
+        pointing_alt = dl1_params["pointing_alt"][index]
+        pointing_alt_deg = np.rad2deg(pointing_alt)
+        zenith = 90.0 - pointing_alt_deg
```

Comment:
since you do not use pointing_alt_deg anywhere else I could combine this and previous line

Reply:
done